lib/circuitry/provisioning/queue_creator.rb

```diff
@@ -0,0 +1,52 @@
+require 'circuitry/services/sqs'
+require 'circuitry/queue'
+
+module Circuitry
+  module Provisioning
+    class QueueCreator
+      include Services::SQS
+
+      attr_reader :queue_name
+
+      def self.find_or_create(queue_name, dead_letter_queue_name: nil, max_receive_count: 8 )
+        creator = new(queue_name)
+        result = creator.create_queue
+        creator.create_dead_letter_queue(dead_letter_queue_name, max_receive_count) if dead_letter_queue_name
+        result
+      end
+
+      def initialize(queue_name)
+        @queue_name = queue_name
```

Comment:
For consistency, I'd rather we add a private `attr_writer`, then use `self.queue_name=` here instead.  I generally prefer this over setting attributes since it helps to catch issues sooner with typos or renaming of attrs.